core/src/main/java/org/elasticsearch/search/SearchHit.java

```diff
@@ -408,7 +418,7 @@ public XContentBuilder toInnerXContent(XContentBuilder builder, Params params) t
             nestedIdentity.toXContent(builder, params);
         } else {
             if (index != null) {
-                builder.field(Fields._INDEX, index);
+                builder.field(Fields._INDEX, RemoteClusterAware.buildRemoteIndexName(clusterAlias, index));
```

Comment:
yeah that was my plan actually. :) I checked the code to make sure this happens?!